### PR TITLE
setUpAll/tearDownAll for performance reasons

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

tests seem to get to slow during the coverage tests. 

- [x] :hankey: Bugfix

### CODE REVIEW

File: tests/js/client/shell/shell-iterators.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ function IteratorSuite(permuteConfigs) {
   };
 
   return {
-    setUp: function () {
+    setUpAll: function () {
```
Comment (jsteemann) [inline tests/js/client/shell/shell-iterators.js:87]: This is not safe.
There are tests in here that **modify** data in the the collection, e.g.
https://github.com/arangodb/arangodb/pull/17322/files#diff-3588a0693f1c76defd4f9f07238e7e37fca75b717e998352ac9c1fd71f21ca1aR226

**Review by jsteemann — CHANGES_REQUESTED** [review]

File: tests/js/client/shell/shell-iterators-read-standalone.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    query: (...args) => internal.db._query(...args),
+    collection: (name) => internal.db[name],
+    abort: () => {},
+  };
+  const permute = function(run) {
+    [ {}, {intermediateCommitCount: 111} ].forEach((opts) => {
+      run(ctx, opts);
+    });
+  };
+  global.read = true;
```
Comment (jsteemann) [inline tests/js/client/shell/shell-iterators-read-standalone.js:49]: What is the purpose of this global? I tried to search for it in this PR, but didn't find any clues. It seems it is just set here, but not read back?

File: tests/js/client/shell/shell-iterators-write-standalane.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    query: (...args) => internal.db._query(...args),
+    collection: (name) => internal.db[name],
+    abort: () => {},
+  };
+  const permute = function(run) {
+    [ {}, {intermediateCommitCount: 111} ].forEach((opts) => {
+      run(ctx, opts);
+    });
+  };
+  global.read = true;
```
Comment (jsteemann) [inline tests/js/client/shell/shell-iterators-write-standalane.js:49]: What is the purpose of this global? I tried to search for it in this PR, but didn't find any clues. It seems it is just set here, but not read?

File: tests/js/client/shell/shell-iterators-read-standalone.js (hunk 2 above)
Comment (dothebart) [reply tests/js/client/shell/shell-iterators-read-standalone.js:49]: will delete. leftover of tries.

**Review by jsteemann — APPROVED** [review]
LGTM

**Pull request merged**